exchanges/poloniex/poloniex_wrapper.go

```diff
@@ -589,8 +590,78 @@ func (p *Poloniex) CancelAllOrders(_ *order.Cancel) (order.CancelAllResponse, er
 
 // GetOrderInfo returns order information based on order ID
 func (p *Poloniex) GetOrderInfo(orderID string, pair currency.Pair, assetType asset.Item) (order.Detail, error) {
-	var orderDetail order.Detail
-	return orderDetail, common.ErrNotYetImplemented
+	orderInfo := order.Detail{
+		Exchange: p.Name,
+		Pair:     pair,
+	}
+
+	trades, err := p.GetAuthenticatedOrderTrades(orderID)
+	if err == nil {
+		for _, o := range trades {
+			var tradeHistory order.TradeHistory
+			tradeHistory.Exchange = p.Name
+			tradeHistory.Side, err = order.StringToOrderSide(o.Type)
+			if err != nil {
+				return orderInfo, err
+			}
+			tradeHistory.TID = strconv.FormatInt(o.GlobalTradeID, 10)
+			tradeHistory.Timestamp, err = time.Parse(common.SimpleTimeFormat, o.Date)
+			if err != nil {
+				return orderInfo, err
+			}
+			tradeHistory.Price = o.Rate
+			tradeHistory.Amount = o.Amount
+			tradeHistory.Total = o.Total
+			tradeHistory.Fee = o.Fee
+
+			orderInfo.Trades = append(orderInfo.Trades, tradeHistory)
+		}
+	}
+
+	resp, err := p.GetAuthenticatedOrderStatus(orderID)
+	if err != nil {
+		return orderInfo, err
+	}
+
+	switch resp.Success {
+	case 0: // fail
+		var errMsg GenericResponse
+		err = json.Unmarshal(resp.Result, &errMsg)
+		if err != nil {
+			return orderInfo, err
+		}
+		if len(orderInfo.Trades) > 0 { // on closed orders return trades only
+			return orderInfo, nil
+		}
+		return orderInfo, fmt.Errorf(errMsg.Error)
+	case 1: // success
+		var status map[string]wsStatus
+		err = json.Unmarshal(resp.Result, &status)
+		if err != nil {
+			return orderInfo, err
+		}
+
+		for _, s := range status {
+			orderInfo.Status, _ = order.StringToOrderStatus(s.Status)
+			orderInfo.Price = s.Rate
+			orderInfo.Amount = s.Amount
+			orderInfo.Cost = s.Total
+			orderInfo.Fee = s.Fee
+			orderInfo.TargetAmount = s.StartingAmount
+
+			orderInfo.Side, err = order.StringToOrderSide(s.Type)
+			if err != nil {
+				return orderInfo, err
+			}
+
+			orderInfo.Date, err = time.Parse(common.SimpleTimeFormat, s.Date)
+			if err != nil {
+				return orderInfo, err
+			}
+		}
+	}
```

Comment:
Can we please move this into GetAuthenticatedOrderStatus? This would be easier for people who might use this package as a library.